pkg/controller/jobs/job/job_controller_test.go

```diff
@@ -384,15 +377,14 @@ func TestPodSets(t *testing.T) {
 					PodAnnotation(kueuealpha.PodSetPreferredTopologyAnnotation, "cloud.com/block").
 					Obj(),
 			),
-			wantPodSets: []kueue.PodSet{{
-				Name: kueue.DefaultPodSetName,
-				Template: jobTemplate.Clone().
-					PodAnnotation(kueuealpha.PodSetPreferredTopologyAnnotation, "cloud.com/block").
-					Spec.Template,
-				Count: 3,
-				TopologyRequest: &kueue.PodSetTopologyRequest{Preferred: ptr.To("cloud.com/block"),
-					PodIndexLabel: ptr.To(batchv1.JobCompletionIndexAnnotation)},
-			}},
+			wantPodSets: []kueue.PodSet{
+				*utiltesting.MakePodSet(kueue.DefaultPodSetName, 3).
+					PodSpec(jobTemplate.Clone().Spec.Template.Spec).
+					Annotations(map[string]string{kueuealpha.PodSetPreferredTopologyAnnotation: "cloud.com/block"}).
+					PreferredTopologyRequest("cloud.com/block").
```

Comment:
Actually, I think `PreferredTopologyRequest` could add the annotation `PodSetPreferredTopologyAnnotation` too, because they are dependent on one-another. WDYT? In any case it can be a follow up.

Reply:
Thanks, great recommendation. I think so, too. Let's refactor the`PreferredTopologyRequest` and `RequiredTopologyRequest`.